src/overlays/actors/ovl_Shot_Sun/z_shot_sun.c

```diff
@@ -32,27 +30,155 @@ const ActorInit Shot_Sun_InitVars = {
+                    this->actor.home.pos.z, 0, 0, 0, fairyType);
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void ShotSun_TriggerFairy(ShotSun* this, PlayState* play) {
+    if ((this->actor.cutscene == -1) || ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
+        if (this->actor.cutscene != -1) {
+            ActorCutscene_Start(this->actor.cutscene, &this->actor);
+        }
+        this->actionFunc = ShotSun_SpawnFairy;
+        this->timer = 50;
+        Actor_Spawn(&play->actorCtx, play, ACTOR_DEMO_KANKYO, this->actor.home.pos.x, this->actor.home.pos.y,
+                    this->actor.home.pos.z, 0, 0, 0, 0x11);
+        Audio_PlaySfxAtPos(&this->actor.projectedPos, NA_SE_EV_TRE_BOX_APPEAR);
+    } else {
+        ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+    }
+}
+
+void func_809738D0(ShotSun* this, PlayState* play) {
+    s32 params = SHOTSUN_GET_FF(&this->actor);
+
+    if (play->msgCtx.ocarinaMode == 3) {
+        switch (play->msgCtx.unk1202E) {
+            case 10:
+                if (params == 0x41) {
+                    this->actionFunc = ShotSun_TriggerFairy;
+                    play->msgCtx.ocarinaMode = 4;
+                }
+                break;
+            case 11:
+                if (params == 0x40) {
+                    this->actionFunc = ShotSun_TriggerFairy;
+                    play->msgCtx.ocarinaMode = 4;
+                }
+                break;
+        }
+    }
+}
+
+void ShotSun_UpdateHyliaSun(ShotSun* this, PlayState* play) {
+    Vec3s cylinderPos;
+    Player* player = GET_PLAYER(play);
+    EnItem00* collectible;
+    s32 pad;
+    Vec3f spawnPos;
+
+    if (this->collider.base.acFlags & AC_HIT) {
+        play_sound(NA_SE_SY_CORRECT_CHIME);
+        if (1) {}
+        if (INV_CONTENT(ITEM_ARROW_FIRE) == ITEM_NONE) {
+            Actor_Spawn(&play->actorCtx, play, ACTOR_ITEM_ETCETERA, 700.0f, -800.0f, 7261.0f, 0, 0, 0,
+                        ITEM_ETC_ARROW_FIRE);
+            if (1) {}
+            gSaveContext.cutsceneTrigger = 1;
+        } else {
+            spawnPos.x = 700.0f;
+            spawnPos.y = -800.0f;
+            spawnPos.z = 7261.0f;
+            collectible = (EnItem00*)Item_DropCollectible(play, &spawnPos, ITEM00_MAGIC_LARGE);
+            if (collectible != NULL) {
+                collectible->unk152 = 0x1770;
+                collectible->actor.speedXZ = 0.0f;
+            }
+        }
+
+        Actor_MarkForDeath(&this->actor);
+    } else if (!(this->actor.xzDistToPlayer > 120.0f)) {
+        if ((gSaveContext.save.time >= CLOCK_TIME(6, 30)) && (gSaveContext.save.time < CLOCK_TIME(7, 30))) {
+            cylinderPos.x = player->bodyPartsPos[7].x + (play->envCtx.unk_4 * 0.16666667f);
+            cylinderPos.y = (player->bodyPartsPos[7].y - 30.0f) + (play->envCtx.unk_8 * 0.16666667f);
+            cylinderPos.z = player->bodyPartsPos[7].z + (play->envCtx.unk_C * 0.16666667f);
```

Comment:
Would those match with `/ 6.0f` or `* (1.0f / 6.0f)`?
```suggestion
            cylinderPos.x = player->bodyPartsPos[7].x + (play->envCtx.unk_4 / 6.0f);
            cylinderPos.y = (player->bodyPartsPos[7].y - 30.0f) + (play->envCtx.unk_8 / 6.0f);
            cylinderPos.z = player->bodyPartsPos[7].z + (play->envCtx.unk_C / 6.0f);
```

Reply:
* (1.0f / 6.0f) worked